### PR TITLE
Prepare jquery mobile work with new version jquery

### LINKED ISSUE #33312: [Issue] Prepare jquery mobile work with new version jquery
This issue is automatically created based on existing pull request: magento/magento2#33107: Prepare jquery mobile work with new version jquery

---------

Fix Uncaught TypeError: Cannot read property 'concat' of undefined in jquery.mobile.custom in jquery 3.6.0

Additional also reduce define deps jquery.mobile.custom in requirejs for frontend scope. We already defined it once as dependency at file lib/mage/menu.js

### Description (*)

This fix for current version magento jquery.mobile.custom can work with higher jquery version. Also this patch can workable with lower jquery as well (backward compatibility)

additional info related
https://www.w3schools.com/jsref/obj_mouseevent.asp

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. This error with jquery.mobile.custom when we use higher version of jquery.
2. Open browser to see error in console tab
3. Make sure in mobile web still work without any issue

### Questions or comments

@sivaschenko @DanielRuf  you may interesting with such changes
Feel free to give me any feedbacks

https://jquerymobile.com/download/
Current jquery mobile latest version stable is 1.4.5. Magento used custom version at 1.4.3. Should we keep use this widget for next version ?

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by mrtuvn**
Fix Uncaught TypeError: Cannot read property 'concat' of undefined in jquery.mobile.custom in jquery 3.6.0

1. Additional also reduce define deps jquery.mobile.custom in requirejs for frontend scope. We already defined it once as dependency at file lib/mage/menu.js
2. Remove jquery.mobile.custom from code base due update new jquery version in future. We eliminated jquery mobile for loose coupling with this oldy friend 

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. This error with jquery.mobile.custom when we use higher version of jquery.
2. Open browser to see error in console tab
3. Make sure in mobile web still work without any issue

### Questions or comments

@sivaschenko @DanielRuf  you may interesting with such changes
Feel free to give me any feedbacks

https://jquerymobile.com/download/
Current jquery mobile latest version stable is 1.4.5. Magento used custom version at 1.4.3. Should we keep use this widget for next version ?

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#33312: Prepare jquery mobile work with new version jquery

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by ihor-sviziev** [comment]
@mrtuvn, your changes look good to me. That's really good to see that we can get rid of jquery mobile.

Q1: Could you provide some description and steps to reproduce?
Q2: Are there any other places that have dependencies on jquery mobile? If no - maybe we can just remove it?

**Comment by mrtuvn** [comment]
as my search usages i only see magento use (jquery.mobile.custom) in lib/mage/menu.js. Also frontend area always request this file as you can see in requirejs-config.js (deps attribute)

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9106](https://jira.corp.magento.com/browse/ENGCOM-9106) has been created to process this Pull Request

**Comment by Den4ik** [comment]
@mrtuvn I see that jquery `events` property is not equal with `virtualEventNames`. Are you sure that it's correct to use this.

Also jq mobile framework very old and don't support past 2 years. May be we should look at replace for this framework

**Comment by ihor-sviziev** [comment]
@mrtuvn is there any benefit that brings jquery mobile in 2021?

**Comment by mrtuvn** [comment]
i think we should not rely on it. Better try to decouple this one

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run WebAPI Tests, Integration Tests, Functional Tests EE, Functional Tests B2B

**Comment by Den4ik** [comment]
@magento run Integration Tests, Functional Tests EE, Functional Tests B2B

**Comment by Den4ik** [comment]
@mrtuvn Could you please check failed integration test? Looks like it's related to changes

**Comment by mrtuvn** [comment]
Maybe module page builder used this event click defined in here. When this file not available the issue will happen
https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/jquery.mobile.custom.js#L465

**Review by Den4ik — CHANGES_REQUESTED** [review]
@mrtuvn Just remove break to new line from end of `dev/tests/integration/testsuite/Magento/Framework/View/_files/static/expected/test.min.js` and Integration Tests should be passed.

Looks like IDE added it automatically

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Functional Tests CE, Functional Tests B2B, Functional Tests EE, Static Tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
sublime-text come to rescue for such case

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Functional Tests B2B, Functional Tests CE

**Comment by Den4ik** [comment]
@magento run Functional Tests EE, Functional Tests B2B, Functional Tests CE

**Comment by sivaschenko** [comment]
Functional test failure is not related to the changes in the PR

**Review by Den4ik — APPROVED** [review]
✅ Approved
Looks good for me.
@mrtuvn thanks for contribution

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9106](https://jira.corp.magento.com/browse/ENGCOM-9106) has been created to process this Pull Request

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9106](https://jira.corp.magento.com/browse/ENGCOM-9106) has been created to process this Pull Request

**Comment by sivaschenko** [comment]
@magento create issue

**Comment by sivaschenko** [comment]
@mrtuvn thanks for the dependency elimination. Can `lib/web/jquery/jquery.mobile.custom.js` be now safely removed from the codebase?

File: lib/web/mage/menu.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ define([
     'jquery',
     'matchMedia',
     'jquery-ui-modules/menu',
-    'jquery/jquery.mobile.custom',
```
Comment (ihor-sviziev) [inline lib/web/mage/menu.js:10]: this file contains "swipeleft" event, that was added in jquery mobile if I'm not mistaken. Is it works now?
I think we can safely remove it. It's not so important at all

Comment (mrtuvn) [reply lib/web/mage/menu.js:10]: i have eliminated it already

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Static Tests, Semantic Version Checker, Integration Tests, Magento Health Index

**Comment by mrtuvn** [comment]
@magento run Database Compare, Sample Data Tests B2B, Sample Data Tests CE, Sample Data Tests EE, Unit Tests, WebAPI Tests

**Comment by mrtuvn** [comment]
@magento give me test instance

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@mrtuvn can we remove the `jquery/jquery.mobile.custom` file completely now?

**Comment by mrtuvn** [comment]
Yes but i'm not sure about other themes or modules still require it or not

**Comment by ihor-sviziev** [comment]
@mrtuvn I think it anyway won't work in case of jquery update, so it's better to remove it

**Comment by Den4ik** [comment]
`jquery/jquery.mobile.custom.js` could be safely removed for magento core. We can't guarantee safety for custom themes. So it should be removed in the same branch where will be updated jquery. If it will be 2.5-develop, then we should not remove it here

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests, Integration Tests, Functional Tests EE

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests look not related to changes from this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9106](https://jira.corp.magento.com/browse/ENGCOM-9106) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
@magento run Integration Tests

**Pull request merged**